source/server/server.cc

```diff
@@ -226,6 +226,9 @@ InstanceUtil::BootstrapVersion InstanceUtil::loadBootstrapConfig(
     MessageUtil::loadFromYaml(config_yaml, bootstrap_override, validation_visitor);
     bootstrap.MergeFrom(bootstrap_override);
   }
+  if (config_proto.ByteSize() != 0) {
+    bootstrap.MergeFrom(config_proto);
```

Comment:
Do you have a server test that actually covers this case?

Reply:
Sorry missed this. Added a few tests to verify the order that Bootstrap config is read from the various config functions.